graph/task.go

```diff
@@ -225,3 +247,45 @@ func getNormalizedDockerImageNames(dockerImages []string, registry string) []str
 
 	return normalizedDockerImages
 }
+
+// the step's environment variables should override the task's default ones if provided
+func mergeEnvs(stepEnvs []string, taskEnvs []string) ([]string, error) {
+	if len(taskEnvs) < 1 {
+		return stepEnvs, nil
+	}
+
+	//preprocess the coma case
+	var newTaskEnvs []string
+	for _, env := range taskEnvs {
+		newEnv := strings.Split(env, ",")
+		newTaskEnvs = append(newTaskEnvs, newEnv...)
+	}
+
+	var stepmap = make(map[string]string)
+	//parse stepEnvs into a map
+	for _, env := range stepEnvs {
+		stringSlice := strings.Split(env, "=")
```

Comment:
How do you handle the value containing `=`? You might want to check helm parser code.

https://github.com/helm/helm/blob/6b222cdc0fc7d94ce22a265d3a3580cc7afd3cab/cmd/helm/install.go#L314